# Урок 2. Дискретные случайные величины. Закон распределения вероятностей. Биномиальный закон распределения. Распределение Пуассона

1.Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
from math import factorial, exp

In [5]:
def bernulli(n, k, p):
  comb = factorial(n)/(factorial(k) * factorial(n-k))
  return comb*(p**k)*(1-p)**(n-k)

In [6]:
print(f'P = {bernulli(100, 85, 0.8):.4f}')

P = 0.0481


P = 0.0481 = 4,81%

2.Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. 
a)Какова вероятность, что ни одна из них не перегорит в первый день? 
b)Какова вероятность, что перегорят ровно две?

In [7]:
def poisson_distr(m, p, n):
  lamb = p * n
  return exp(-lamb)*(lamb**m)/factorial(m)

In [8]:
#a)
print(f'P = {poisson_distr(0, 0.0004, 5000):.4f}')

P = 0.1353


P = 0.1353 = 0,1353%

In [9]:
#b)
print(f'P = {poisson_distr(2, 0.0004, 5000):.4f}')

P = 0.2707


P = 0.2707 = 0,2707%

3.Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [10]:
print(f'P = {bernulli(144, 70, 0.5):.4f}')

P = 0.0628


P = 0.0628 = 6,28%

4.В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. a)Какова вероятность того, что все мячи белые? b)Какова вероятность того, что ровно два мяча белые? c)Какова вероятность того, что хотя бы один мяч белый?

In [12]:
def combinations(n, k):
  return int(factorial(n) / (factorial(k) * factorial(n-k)))

In [13]:
n1 = combinations(10, 2)
n2 = combinations(11, 2)
print(f'first basket = {n1}')
print(f'second basket = {n2}')

first basket = 45
second basket = 55


In [16]:
m1 = combinations(7, 2)
m2 = combinations(9, 2)
print(f'm1 = {m1}')
print(f'm2 = {m2}')

m1 = 21
m2 = 36


In [17]:
print(f'P(A) = {m1/n1:.4f}')
print(f'P(B) = {m2/n2:.4f}')
print(f'P(A+B) = {(m1/n1)*(m2/n2):.4f}')

P(A) = 0.4667
P(B) = 0.6545
P(A+B) = 0.3055


P = 0.3055 = 30,55%

In [19]:
m_A = combinations(7,2)
m_B = combinations(2,2)
m_C = combinations(7,1)*combinations(3,1)
m_D = combinations(9,1)*combinations(2,1)
m_E = combinations(3,2)
m_F = combinations(9,2)

print(f'm_A = {m_A}')
print(f'm_B = {m_B}')
print(f'm_C = {m_C}')
print(f'm_D = {m_D}')
print(f'm_E = {m_E}')
print(f'm_F = {m_F}')

m_A = 21
m_B = 1
m_C = 21
m_D = 18
m_E = 3
m_F = 36


In [20]:
P = (m_A/n1)*(m_B/n2)+(m_C/n1)*(m_D/n2)+(m_E/n1)*(m_F/n2)
print(f'P = {P:.4f}')

P = 0.2048


P = 0.2048 = 20,48%

In [22]:
#c)
m1 = combinations(3, 2)
m2 = combinations(2, 2)
print(f'm1 = {m1}')
print(f'm2 = {m2}')

m1 = 3
m2 = 1


In [23]:
print(f'P(A_) = {(m1/n1)*(m2/n2):.4f}')
print(f'P(A) = {1 - (m1/n1)*(m2/n2):.4f}')

P(A_) = 0.0012
P(A) = 0.9988


P(A) = 0.9988 = 99.88%